libclc/ptx-nvidiacl/libspirv/SPV_EXT_shader_atomic_float_add/atomicfaddext.cl

```diff
@@ -0,0 +1,131 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <spirv/spirv.h>
+#include <spirv/spirv_types.h>
+
+#pragma OPENCL EXTENSION cl_khr_fp64 : enable
```

Comment:
```suggestion
#ifdef cl_khr_fp64
#pragma OPENCL EXTENSION cl_khr_fp64 : enable
#endif // cl_khr_fp64
```
Typical idiom is use `#ifdef cl_khr_fp64` for the code using doubles. I suggest organize the code in the following way:
```C++
// FP32 atomics - must work w/o additional extensions
float __clc__atomic_fetch_*
...
// FP64 atomics - require cl_khr_fp64 extension
#ifdef cl_khr_fp64
#pragma OPENCL EXTENSION cl_khr_fp64 : enable
double __clc__atomic_fetch_*
...
#endif // cl_khr_fp64
// FP16 - some day ???
```

E.g. https://github.com/intel/llvm/blob/sycl/libclc/ptx-nvidiacl/libspirv/relational/isinf.cl

Reply:
Fixed in https://github.com/intel/llvm/pull/3276/commits/611e25ed97298a1bc8bdabfa0338db1811e3f8e6.